editor/src/components/canvas/ui-jsx-canvas.tsx

```diff
@@ -361,29 +370,123 @@ export const UiJsxCanvas = betterReactMemo(
                   updateInvalidatedPaths,
                   shouldIncludeCanvasRootInTheSpy,
                 )
-                const exportsDetail = projectFile.fileContents.parsed.exportsDetail
                 let filteredScope: MapLike<any> = {
                   ...scope.module.exports,
                   __esModule: true,
                 }
-                for (const s of Object.keys(scope)) {
-                  if (s in exportsDetail.namedExports) {
-                    filteredScope[s] = scope[s]
-                  } else if (exportsDetail.defaultExport != null) {
-                    if (
-                      (isExportDefaultNamed(exportsDetail.defaultExport) ||
-                        isExportDefaultModifier(exportsDetail.defaultExport)) &&
-                      s === exportsDetail.defaultExport.name
-                    ) {
-                      filteredScope['default'] = scope[s]
-                    } else if (
-                      isExportDefaultFunction(exportsDetail.defaultExport) ||
-                      isExportDefaultExpression(exportsDetail.defaultExport)
-                    ) {
-                      filteredScope['default'] = scope['default']
-                    }
+
+                function addToFilteredScopeFromSpecificScope(
+                  filteredScopeKey: string,
+                  scopeKey: string,
+                  scopeToWorkWith: MapLike<any>,
+                ): void {
+                  if (scopeKey in scopeToWorkWith) {
+                    filteredScope[filteredScopeKey] = scopeToWorkWith[scopeKey]
                   }
                 }
+
+                function addToFilteredScope(filteredScopeKey: string, scopeKey: string): void {
```

Comment:
can this be moved to be a top level function?

Reply:
It could but the two scopes would have to be pulled up into parameters.

Reply:
I'm just a bit sad that the ui-jsx-canvas render function is becoming this unholy amalgam of module resolution + entry point for the canvas rendering

Reply:
A better solution would be pull the whole block of export handling into a top level function, I _think_ that would work.

Reply:
Fixed in 5591ee12.